opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/NoFieldQuery.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage.script.filter.lucene.relevance;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.sql.common.antlr.SyntaxCheckException;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+
+/**
+ * Base class to represent relevance queries that search multiple fields.
+ *
+ * @param <T> The builder class for the OpenSearch query.
+ */
+abstract class NoFieldQuery<T extends QueryBuilder> extends RelevanceQuery<T> {
+  public NoFieldQuery(Map<String, QueryBuilderStep<T>> queryBuildActions) {
+    super(queryBuildActions);
+  }
+
+  /**
+   * Override build function because RelevanceQuery requires 2 fields,
+   * but NoFieldQuery must have no fields.
+   *
+   * @param func : Contains function name and passed in arguments.
+   * @return : QueryBuilder object
+   */
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    var arguments = func.getArguments().stream().map(
+        a -> (NamedArgumentExpression) a).collect(Collectors.toList());
+    if (arguments.size() < 1) {
+      throw new SyntaxCheckException(String.format(
+          "%s requires at least two parameters", getQueryName()));
+    }
+
+    // Aggregate parameters by name, so getting a Map<Name:String, List>
+    arguments.stream().collect(Collectors.groupingBy(a ->
+        a.getArgName().toLowerCase())).forEach((k, v) -> {
+          if (v.size() > 1) {
+            throw new SemanticCheckException(String.format(
+                "Parameter '%s' can only be specified once.", k));
+          }
+        });
+
+    T queryBuilder = createQueryBuilder(arguments);
+
+    arguments.removeIf(a -> a.getArgName().equalsIgnoreCase("query"));
+
+    var iterator = arguments.listIterator();
+    while (iterator.hasNext()) {
+      NamedArgumentExpression arg = iterator.next();
+      String argNormalized = arg.getArgName().toLowerCase();
+
+      if (!getQueryBuildActions().containsKey(argNormalized)) {
+        // query name should be updated in the exception message
+        // once functions other than query extend NoFieldQuery
+        throw new SemanticCheckException(String.format(
+            "Parameter %s is invalid for query function.", argNormalized));
```

Comment:
This will result in a misleading exception message to the user since we are mapping query function to use query_string.

Reply:
I believe we weren't able to do this because the name returned from OpenSearch was `query_string`, but we wanted the name `query` to be used to remove confusion if someone comes across this error in the future. @GumpacG please confirm/correct my comment if my understanding is wrong.

Reply:
Oh, right. What about this?
```suggestion
            "Parameter %s is invalid for %s function.", argNormalized, func.getFunctionName()));
```

Reply:
Oh, right, but what about this?
```suggestion
            "Parameter %s is invalid for %s function.", argNormalized, func.getFunctionName()));
```

Reply:
This fixes the issue in this case, but causes another test to fail.
The test `throws_SemanticCheckException_when_wrong_argument_name()` in `RelevanceQueryBuildTests.java` fails saying
```
expected:<...is invalid for mock_[query] function.> but was:<...is invalid for mock_[function] function.>
Expected :Parameter wrongarg is invalid for mock_query function.
Actual   :Parameter wrongarg is invalid for mock_function function.
```
The test seems to use the query name instead of the function name

Reply:
Fixed in 8f1e7f775a4b3b203a754745d3fd421b56294f59